From 0784983e11d7bc1d8f28420ee061eece521a9590 Mon Sep 17 00:00:00 2001
From: Eric Jarrige <eric.jarrige@armadeus.org>
Date: Mon, 23 May 2016 12:55:20 +0200
Subject: [PATCH 5/9] imx27: Dcache is not fully flushed during the boot
 process.

This patch fix it by changing the "test and clean DCache" for a
"test, clean, and invalidate DCache" instruction set and make it possible to run
U-Boot from RAM.

Tested method: start U-Boot from RAM with a go a0000800"

Signed-off-by: Eric Jarrige <eric.jarrige@armadeus.org>
---
 arch/arm/cpu/arm926ejs/start.S | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/cpu/arm926ejs/start.S b/arch/arm/cpu/arm926ejs/start.S
index 959d1ed..339a8bd 100644
--- a/arch/arm/cpu/arm926ejs/start.S
+++ b/arch/arm/cpu/arm926ejs/start.S
@@ -77,7 +77,7 @@ cpu_init_crit:
 	 */
 	mov	r0, #0
 flush_dcache:
-	mrc	p15, 0, r15, c7, c10, 3
+	mrc	p15, 0, r15, c7, c14, 3
 	bne	flush_dcache
 
 	mcr	p15, 0, r0, c8, c7, 0	/* invalidate TLB */
-- 
2.9.3

